c7n/policy.py

```diff
@@ -802,6 +802,35 @@ def validate(self):
                 'policy:%s resource:%s does not have a cloudformation type'
                 ' and is there-fore not supported by config-poll-rule'))
 
+    def get_obsolete_evaluations(self, client, cfg_rule_name, evaluations):
+        """Get list of evaluations that are no longer applicable due to resources being deleted
+        """
+        latest_resource_ids = set()
+        for latest_eval in evaluations:
+            latest_resource_ids.add(latest_eval['ComplianceResourceId'])
+
+        obsolete_evaluations = []
+        paginator = client.get_paginator('get_compliance_details_by_config_rule')
+        old_evals = paginator.paginate(
+            ConfigRuleName=cfg_rule_name,
+            ComplianceTypes=['COMPLIANT', 'NON_COMPLIANT'],
+            PaginationConfig={'PageSize': 100}).build_full_result().get('EvaluationResults', [])
+
+        for old_eval in old_evals:
+            old_resource_id = old_eval['EvaluationResultIdentifier']
+            ['EvaluationResultQualifier']['ResourceId']
```

Comment:
@kk1532 i got a warning on this line
```
c7n/policy.py:821
  /Users/ddao/git/kk1532-cloud-custodian/c7n/policy.py:821: SyntaxWarning: list indices must be integers or slices, not str; perhaps you missed a comma?
    ['EvaluationResultQualifier']['ResourceId']
```

I think it's because how you're breaking the code into multiple lines. I think it's better if we do something like this instead
```
        for old_eval in old_evals:
            eval_res_qual = old_eval['EvaluationResultIdentifier']['EvaluationResultQualifier']
            old_resource_id = eval_res_qual['ResourceId']
            if old_resource_id not in latest_resource_ids:
                obsolete_evaluation = {
                    'ComplianceResourceType': eval_res_qual['ResourceType'],
                    'ComplianceResourceId': old_resource_id,
                    'Annotation': 'The rule does not apply.',
                    'ComplianceType': 'NOT_APPLICABLE',
                    'OrderingTimestamp': datetime.now()}
                obsolete_evaluations.append(obsolete_evaluation)
```
cleaner that way